src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -806,3 +809,20 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
         }
     # endregion
 
+    def __removed_older_complete_status_files(self, status_folder):
+        """ Retain 10 latest status complete file and remove other .complete.status files """
+        files_to_removed = []
+        complete_status_files_list = glob.glob(status_folder + '/' + '*.complete.status')   # Glob return empty list if no file matched pattern
+        if len(complete_status_files_list) <= Constants.MAX_COMPLETE_STATUS_FILES_COUNT:    # 9 + 1 complete_status_file to be created
```

Comment:
nit: 

1. My understanding is Constants.MAX_COMPLETE_STATUS_FILES_COUNT should refer to the max # of files allowed to exist on disk (this includes the new one to be created). Because for any debugging or reference, we will always refer to the value of MAX_COMPLETE_STATUS_FILES_COUNT in Constants. So Constants file should have the final and correct value which in this case is 10.

To handle this here, keep the value in Constants as '10', and modify the condition here to:
if len(complete_status_files_list) < Constants.MAX_COMPLETE_STATUS_FILES_COUNT:


2. Or if we want to keep the existing implementation, consider renaming the constant to MAX_COMPLETE_STATUS_FILES_TO_RETAIN

Reply:
i use the MAX_COMPLETE_STATUS_FILES_TO_RETAIN with value of 10, it took me  some time to figure out that 1.complete .status might get deleted because it's the first to be created and modify time will be oldest. and when it does get deleted the load_status will recreate it again, at this time the complete_status_files_list will again have 1.complete.status, so there's an additional file count.